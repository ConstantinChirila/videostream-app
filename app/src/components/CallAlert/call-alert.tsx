import React, { ReactElement } from "react";

type TProps = {
  caller: string;
  acceptCall: () => void;
};

export function CallAlert({ caller, acceptCall }: TProps): ReactElement {
  return (
    <div>
      <h1>{caller} is calling you</h1>
      <button onClick={acceptCall}>Accept</button>
    </div>
  );
}
