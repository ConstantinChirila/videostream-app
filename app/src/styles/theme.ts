export const theme = {
  colors: {
    primary: "#FF286A",
    secondary: "#5000B4",
    dark: "#172029",
    light: "#F5F6F8",
  },
} as any;
