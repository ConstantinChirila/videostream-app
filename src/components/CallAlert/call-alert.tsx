import React, { ReactElement } from "react";
import { StyledContainer, StyledHeading, StyledButton } from "./";

type TProps = {
  caller: string;
  acceptCall: () => void;
};

export function CallAlert({ caller, acceptCall }: TProps): ReactElement {
  return (
    <StyledContainer>
      <StyledHeading>{caller} is calling you</StyledHeading>
      <StyledButton type="button" onClick={acceptCall}>
        Accept
      </StyledButton>
    </StyledContainer>
  );
}
