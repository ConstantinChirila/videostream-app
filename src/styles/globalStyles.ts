import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";

export const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;1,500&display=swap');

  html {
    font-size: 10px;
  }
  
  body {
    font-family: 'Roboto', sans-serif;
    color: ${theme.colors.dark};
    background-color: ${theme.colors.light};
    font-size: 1.6rem
  }
`;
