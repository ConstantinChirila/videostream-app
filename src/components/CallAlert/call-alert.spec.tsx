import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styles";
import { CallAlert } from "./call-alert";

describe("CallAlert", () => {
  it("should render the component", () => {
    const mockFn = jest.fn();
    const { container } = render(
      <ThemeProvider theme={theme}>
        <CallAlert caller="test" acceptCall={mockFn} />
      </ThemeProvider>
    );
    const heading = screen.getByText(/test is calling you/i);
    expect(heading).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should accept the call", () => {
    const mockFn = jest.fn();
    render(
      <ThemeProvider theme={theme}>
        <CallAlert caller="test" acceptCall={mockFn} />
      </ThemeProvider>
    );

    fireEvent.click(screen.getByText(/Accept/i));
    expect(mockFn).toHaveBeenCalled();
  });
});
