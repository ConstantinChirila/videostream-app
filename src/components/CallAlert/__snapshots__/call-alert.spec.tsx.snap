// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallAlert should render the component 1`] = `
<div
  class="sc-bdfBwQ kGxswN"
>
  <h1
    class="sc-gsTCUz eqiecY"
  >
    test
     is calling you
  </h1>
  <button
    class="sc-hKgILt eUJoKM"
    type="button"
  >
    Accept
  </button>
</div>
`;
